### ISSUE TITLE
mssql v1.18.0 `mssql: SQL Tools Service component` could not start. Fails to start due to `AzureResourceProvider` crashing ?

### ISSUE THREAD

**Issue by gdvallance** [microsoft/vscode-mssql#17648]
- MSSQL Extension Version: 1.18.0
- VSCode Version: 1.76.2
- OS Version: Windows_NTx64 10.0.19042

Steps to Reproduce:

1. Try using `vscode-mssql` extension and on starting it crashes with following:

```text
mssql: SQL Tools Service component could not start  
Source: SQL Server (mssql) (Extension)
```
 &

```text
The AzureResourceProvider server crashed 5 times in the last 3 minutes. The server will not be restarted
Source: SQL Server (mssql) (Extension)

```

& (in the Output tab) -- abridged to remove my-user-account

```text
Failed to load the dll from [C:\Users\my-user-name-\.vscode\extensions\ms-mssql.mssql-1.18.0\sqltoolsservice\4.5.1.6\Windows\coreclr.dll], HRESULT:   0x800700C1
[Info  - 1:12:23 PM] Connection to server got closed. Server will restart.
Failed to load the dll from [C:\Users\my-user-name-\.vscode\extensions\ms-mssql.mssql-1.18.0\sqltoolsservice\4.5.1.6\Windows\coreclr.dll], HRESULT: 0x800700C1
[Info  - 1:12:26 PM] Connection to server got closed. Server will restart.
Failed to load the dll from [C:\Users\my-user-name-\.vscode\extensions\ms-mssql.mssql-1.18.0\sqltoolsservice\4.5.1.6\Windows\coreclr.dll], HRESULT: 0x800700C1
[Info  - 1:12:28 PM] Connection to server got closed. Server will restart.
Failed to load the dll from [C:\Users\my-user-name-\.vscode\extensions\ms-mssql.mssql-1.18.0\sqltoolsservice\4.5.1.6\Windows\coreclr.dll], HRESULT: 0x800700C1
[Info  - 1:12:30 PM] Connection to server got closed. Server will restart.
Failed to load the dll from [C:\Users\my-user-name-\.vscode\extensions\ms-mssql.mssql-1.18.0\sqltoolsservice\4.5.1.6\Windows\coreclr.dll], HRESULT: 0x800700C1
[Error - 1:12:33 PM] Connection to server got closed. Server will not be restarted.

```

- I can confirm that the file in question is there.
- I have restarted the extension
- I have installed and reinstalled the extension

- Anyone know what is going on?

**Comment by gdvallance** [comment]
Reverting the extension to the previous version restored functionality. There is definitely an issue with the v1.18.0 release cf. v1.17.1 release.

**Comment by cheenamalhotra** [comment]
@gdvallance 

Can you attach the files with below info:
1. Make sure you have enabled 'Verbose' logging, with setting: `"mssql.tracingLevel": "Verbose"`
2. Run Command Pallete (Ctrl+Shift+P) > `Developer: Open Logs Folder`
3. Navigate to path: `\window1\exthost\ms-mssql.mssql\` > Copy both files from here.
4. Copy file from path: `\window1\exthost\output_logging_******\#-MSSQL.log`

Attach the 3 files copied here.

**Comment by cheenamalhotra** [comment]
P.S. I was able to validate the extension works fine on Windows 10 Version 10.0.19044 using below VS Code version:

Version: 1.77.0 (user setup)
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T10:02:16.981Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

**Comment by cheenamalhotra** [comment]
Closing as not enough details available to investigate issue

**Comment by davidpcaldwell** [comment]
Hi @cheenamalhotra -- I just wanted to mention that on macOS 13.5 (on an M1 machine), we had the exact same symptoms reported here. We tried some of your diagnostics, but were not able to figure anything out.

*However*, like the original poster, when we downgraded the extension to 1.17.1, it started working. So our experience was that something is in fact wrong with 1.18.0 (and still with the current version, which IIRC is 1.20.1).
